hbjs.js

```diff
@@ -533,15 +533,34 @@ function hbjs(Module) {
     return trace;
   }
 
+  function get_version() {
+    var major = exports.malloc(4);
+    var minor = exports.malloc(4);
+    var patch = exports.malloc(4);
+    heapu32[major / 4] = 0;
+    heapu32[minor / 4] = 0;
+    heapu32[patch / 4] = 0;
+    exports.hb_version(major, minor, patch);
+    let hbversion =
+      heapu32[major / 4] + "." + heapu32[minor / 4] + "." + heapu32[patch / 4];
+    exports.free(major);
+    exports.free(minor);
+    exports.free(patch);
```

Comment:
If you are returning a string anyway, you can use `hb_version_string()`.

Reply:
You can, but I struggled to get WASM to convert the `const char*` pointer from `hb_version_string` into anything useful; it's not the same as just writing into a provider `char *`.

Reply:
```
    var versionPtr = exports.hb_version_string();
    var version = utf8Decoder.decode(heapu8.subarray(versionPtr, heapu8.indexOf(0, versionPtr)));
    return version;
```

(see `trace.push` few lines above)